### ISSUE TITLE
Verify that '#pragma once' is an acceptable header guard lock

### ISSUE THREAD

**Issue by Savagedlight** [Savagedlight/libifconfig#3]
Check whether all compilers in base supports the '#pragma once' guard lock. Do also consider corner cases, like multiple copies of the header being included, from different paths.

Reported by: Ravi Pokala

**Comment by asomers** [comment]
Nope.  While the base compilers all technically support `#pragma once`, it is considered bad style.  Except for a few things in contrib, no header file in the base system uses it.  Better to use the usual `ifndef IFCONFIG_H` style.
